packages/node/src/indexer/poi.service.ts

```diff
@@ -1,71 +1,62 @@
 // Copyright 2020-2021 OnFinality Limited authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
-import { OnApplicationShutdown } from '@nestjs/common';
+import { Inject, Injectable, OnApplicationShutdown } from '@nestjs/common';
 import { u8aConcat, numberToU8a, u8aToBuffer } from '@polkadot/util';
 import { blake2AsHex } from '@polkadot/util-crypto';
+import { QueryTypes, Sequelize } from 'sequelize';
 import { SubqueryProject } from '../configure/project.model';
+import { SubqueryRepo } from '../entities';
+import {
+  PoiFactory,
+  PoiRepo,
+  ProofOfIndex,
+  PoiModel,
+} from './entities/Poi.entity';
+import { PoiBlock } from './PoiBlock';
 
 const DEFAULT_PARENT_HASH = '0xabcd';
 
+@Injectable()
 export class PoiService implements OnApplicationShutdown {
   private isShutdown = false;
-  private latestParentHash: string;
+  private latestPoiBlockHash: string | Buffer;
   private currentBlockHeight: number;
   private chainBlockHash: string;
   private projectId: string;
   private operationHashRoot: string;
+  private poiRepo: PoiRepo;
 
-  constructor(protected project: SubqueryProject) {}
+  constructor(
+    protected project: SubqueryProject,
+    protected sequelize: Sequelize,
+  ) {}
 
   onApplicationShutdown(): void {
     this.isShutdown = true;
   }
 
-  register(
-    id: number,
-    chainBlockHash: string,
-    operationHashRoot: string,
-    projectId: string,
-  ) {
-    this.currentBlockHeight = id;
-    this.chainBlockHash = chainBlockHash;
-    this.operationHashRoot = operationHashRoot;
-    this.projectId = projectId;
-  }
-
-  generateBlockHash(): string {
-    if (
-      !this.currentBlockHeight ||
-      !this.chainBlockHash ||
-      !this.operationHashRoot ||
-      !this.projectId
-    ) {
-      throw new Error('Poof of index: can not generate block hash');
+  async getLatestPoiBlockHash(schema?: string): Promise<string | Buffer> {
+    if (schema) {
+      this.poiRepo = PoiFactory(this.sequelize, schema);
+      const lastPoi = await this.poiRepo.findOne({
+        order: [['createdAt', 'DESC']],
+      });
+      if (lastPoi === null) {
+        this.latestPoiBlockHash = DEFAULT_PARENT_HASH;
+      } else if (lastPoi !== null && lastPoi.hash) {
```

Comment:
`poiRepo.findOne()` doesn't suppose to return `PoiBlock` object. so lastPoi.hash won't work.